Automattic/wp-calypso
# Blog onboarding: Add choose domain task

Related to 2218-gh-Automattic/dotcom-forge

## Proposed Changes

* Add `Choose a domain` flow
* Once user chosse a plan, both domain + plan are added to cart: 

https://user-images.githubusercontent.com/1044309/235729893-e1b68f53-e8a9-40ab-a99b-84715bca8a5e.mp4



## Testing Instructions



* Access an unlaunched site Launchpad in the new write flow: `http://calypso.localhost:3000/setup/start-writing/launchpad?siteSlug=unlaunchedsite22.wordpress.com&start-writing=true`
* You should be able to `Choose a domain` and once you do it, check if that domain is selected in the launchpad and also marked as done
* After finishing the `Choose a plan` step, you should see both products in the cart


## Useful commands (wpsh)

- Left the first one dashed: `update_blog_option(blogId, 'launchpad_checklist_tasks_statuses', ['first_post_published' => true])`
- Clear sidebar items: `delete_blog_option(blogId, 'launchpad_checklist_tasks_statuses')`

## Known issue

We have a React warning while selecting a domain name. We'll fix that in this follow-up task (2296-gh-Automattic/dotcom-forge).

## Pre-merge Checklist
